keyform/views.py

```diff
@@ -147,6 +147,15 @@ class NewContactView(LoginRequiredMixin, PermissionRequiredMixin, CreateView):
     permission_required = 'keyform.add_contact'
 
 
+
+    def form_valid(self, form):
+        if Contact.objects.filter(email__iexact=form.instance.email).exists():
+            messages.error(self.request, 'Error: contact already exists, you have been redirected to edit')
+            return HttpResponseRedirect(reverse('edit-contact', kwargs={'pk':Contact.objects.filter(email__iexact=form.instance.email)[0].pk}))
+
+        return super().form_valid(form)
```

Comment:
Since we expect contacts in the database to be unique we should add that constraint in the models. Here is an equivalent redirect code.

```python
    def form_invalid(self, form):
        if list(filter(lambda e: e.code == 'unique', form.errors.as_data()['email'])):
            messages.error(self.request, 'Error: This contact already exists, you have been redirected to it.')
            preexisting_contact = Contact.objects.filter(email__iexact=form.instance.email).get()
            return HttpResponseRedirect(reverse('edit-contact', kwargs={'pk': preexisting_contact.pk}))

        return super().form_invalid(form)
```